### PR TITLE
test: increase transfer test coverage

### PR DESCRIPTION
**Pull request by steffenix**
@fubuloubu does that look like what you need help on regarding increasing test coverage on #158 ?

### CODE REVIEW

File: contracts/test/TokenWithBlackList.sol
Comment (fubuloubu) [inline contracts/test/TokenWithBlackList.sol:26]: Can we add this funxtionality to our test token instead? (Rename to `public mapping (address => bool) _blocked` and `_setBlocked(address, bool)`)

Comment (fubuloubu) [reply contracts/test/TokenWithBlackList.sol:26]: No reason to create an entire new token for one testing feature

File: tests/functional/vault/test_withdrawal.py
Comment (fubuloubu) [inline tests/functional/vault/test_withdrawal.py:285]: With the change above, this is now what the test can be refactored to

File: contracts/test/TokenWithBlackList.sol
Comment (steffenix) [reply contracts/test/TokenWithBlackList.sol:26]: Sure I will do it

File: tests/functional/vault/test_withdrawal.py
Comment (steffenix) [reply tests/functional/vault/test_withdrawal.py]: I do not see suggested changes.

File: tests/functional/vault/test_withdrawal.py
Comment (fubuloubu) [inline tests/functional/vault/test_withdrawal.py:264]: Not necessary if merged into previous test

File: tests/functional/vault/test_misc.py
Comment (fubuloubu) [reply tests/functional/vault/test_misc.py:142]: Not sure I understand why the comment `Need a fresh vault to do this math right` is there

File: tests/functional/vault/test_withdrawal.py
Comment (steffenix) [reply tests/functional/vault/test_withdrawal.py:264]: Understood!

File: tests/functional/vault/test_withdrawal.py
Comment (fubuloubu) [inline tests/functional/vault/test_withdrawal.py:283]: This entire test should go away now that you merged the functionality into the other one

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
Awesome, will merge once tests pass

**Comment by steffenix** [comment]
Do you need multiple approval to merge?

**Pull request merged**
